intlogger.go

```diff
@@ -93,17 +93,21 @@ func newLogger(opts *LoggerOptions) *intLogger {
 		level = DefaultLevel
 	}
 
-	mutex := opts.Mutex
-	if mutex == nil {
-		mutex = new(sync.Mutex)
+	locker := opts.Locker
+	if locker == nil {
+		mutex := opts.Mutex
+		if mutex == nil {
+			mutex = new(sync.Mutex)
+		}
+		locker = DefaultLocker{Mutex: mutex}
```

Comment:
Is there a reason to introduce `DefaultLocker`? A mutex already satisfies the interface.